src/service_manager.rs

```diff
@@ -208,4 +209,41 @@ impl ServiceManager {
             Ok(Service::new(unsafe { ScHandle::new(service_handle) }))
         }
     }
+
+    /// Retrieves the service name of the specified service.
+    pub fn get_service_key_name(&self, display_name: impl AsRef<OsStr>) -> Result<OsString> {
+        let service_display_name =
+            WideCString::from_os_str(display_name).map_err(|_| Error::DisplayNameHasNulByte)?;
+        let mut size = 0u32;
+        let res = unsafe {
+            Services::GetServiceKeyNameW(
+                self.manager_handle.raw_handle(),
+                service_display_name.as_ptr(),
+                std::ptr::null_mut(),
+                &mut size as _,
+            )
+        };
+
+        if res == 0 && unsafe { GetLastError() } != ERROR_INSUFFICIENT_BUFFER {
+            return Err(Error::Winapi(io::Error::last_os_error()));
```

Comment:
This effectively calls `GetLastError` twice. I think we should get this down to a single time one way or another. You can store away the error and use `Error::from_raw_os_error` for example. But given the feedback above, this code might no longer be needed at all.

Reply:
Removed